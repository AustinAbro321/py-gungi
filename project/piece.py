from dataclasses import dataclass
from enum import Enum
from .color import Color
from .square import Square


class PieceType(Enum):
    MAJOR_GENERAL = 1
    LIEUTENANT_GENERAL = 2
    GENERAL = 3
    ARCHER = 4
    KNIGHT = 5
    MUSKETEER = 6
    CAPTAIN = 7
    SAMURAI = 8
    FORTRESS = 9
    CANNON = 10
    SPY = 11
    PAWN = 12
    KING = 13


@dataclass
class Piece:
    piece_type: PieceType
    color: Color

class MajorGeneral:
    def available_movements(self,place_in_stack: int) -> 'list[Square]':
        pass

class PieceStack:
    def __init__(self,pieces = []):
        self.pieces = pieces

    def pop(self) -> Piece:
        return self.pieces.pop()

    def add(self,piece:Piece):
        self.pieces.append(piece)

    def __repr__(self) -> str:
        return "[" + ", ".join([piece.piece_type.name for piece in self.pieces]) + "]"
