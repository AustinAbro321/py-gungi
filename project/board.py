from .piece import PieceStack
from .square import Square, A1, B7


def starting_board() -> 'list[list[PieceStack]]':
    return [
        [PieceStack([]),PieceStack([]),PieceStack([]),PieceStack([]),
        PieceStack([]),PieceStack([]),PieceStack([]),PieceStack([]),PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])],
        [PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]),
        PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([]), PieceStack([])]]


class Board:
    def __init__(self, content: 'list[list[PieceStack]]'):
        self.content = content

    def __repr__(self):
        name = ""
        for line in self.content:
            name += "["
            name += ", ".join([pieceStack.__repr__() for pieceStack in line])
            name += "]"
            name += "\n"
        return name

    def get_piece_at(self, square: Square):
        row,col = translate_square_to_tuple(square)
        return self.content[row][col]

def translate_square_to_tuple(square: Square):
    row = (square // 9)
    col = (square % 9)
    return row,col
