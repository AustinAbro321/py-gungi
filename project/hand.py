from piece import Piece, PieceType
from color import Color


def starting_hand(player: Color):
    hand = []
    hand.extend([Piece(PieceType.MAJOR_GENERAL, player)] * 4)
    hand.extend([Piece(PieceType.LIEUTENANT_GENERAL, player)] * 4)
    return hand
    # Piece(PieceType.LIEUTENANT_GENERAL, player), Piece(PieceType.LIEUTENANT_GENERAL, player),
    # Piece(PieceType.LIEUTENANT_GENERAL, player), Piece(PieceType.LIEUTENANT_GENERAL, player),
    # Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player),
    # Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player),
    # Piece(PieceType.Archer, player), Piece(PieceType.Archer, player),
    # Piece(PieceType.Knight, player), ]


class Hand:
    def __init__(self, player: Color):
        self.player = Color.WHITE
        self.hand = starting_hand(player)
        print(starting_hand(player))
