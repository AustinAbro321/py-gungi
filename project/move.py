from dataclasses import dataclass
from .board import Board
from .square import *


@dataclass
class MoveInfo:
    point_a: Square
    point_b: Square

def translate_move() -> MoveInfo:
    return MoveInfo(Square(A1),Square(A2))

def move(board : Board,move_info: MoveInfo) -> Board:
    new_board = board
    new_board.get_piece_at(move_info.point_b) \
        .add(board.get_piece_at(move_info.point_a).pop())
    return new_board
