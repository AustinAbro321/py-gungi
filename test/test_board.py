from approvaltests import verify

from board import starting_board, Board, translate_square_to_tuple
from color import Color
from hand import starting_hand
from move import MoveInfo, move
from square import *
from piece import Piece, PieceType, PieceStack


def test_board_init():
    assert True


def test_color():
    assert Color.WHITE == Color.WHITE


def test_starting_hand():
    assert starting_hand(Color.WHITE) == [Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE)]

def test_move_approvals():
    board = Board(starting_board())
    board.content[0][0] = PieceStack([Piece(PieceType.MAJOR_GENERAL, Color.WHITE)])
    verify(move(board, MoveInfo(Square(A1), Square(B1))))

def test_move_approvals_two_on_stack():
    board = Board(starting_board())
    board.content[0][0] = PieceStack([Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                      Piece(PieceType.MAJOR_GENERAL, Color.WHITE)])
    make_move(board, MoveInfo(Square(A1), Square(B1)))

def make_move(board: Board, move_info):
    verify(move(board, move_info))

def test_get_piece_stack_a1():
    get_piece_stack_given_square(0, 0, A1)

def test_get_piece_stack_b7():
    get_piece_stack_given_square(6,1,B7)

def test_get_piece_stack_f9():
    get_piece_stack_given_square(8,8,F9)

def test_get_piece_stack_d8():
    get_piece_stack_given_square(7,3,D8)

def get_piece_stack_given_square(row: int, col: int,square: Square):
    e_row,e_col = translate_square_to_tuple(square)
    assert e_row == row
    assert e_col == col
