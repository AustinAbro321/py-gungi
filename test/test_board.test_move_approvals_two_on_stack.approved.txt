[[MAJOR_GENERAL], [MAJOR_GENERAL], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
[[], [], [], [], [], [], [], [], []]
